symbolic-symcache/src/writer.rs

```diff
@@ -381,8 +381,16 @@ where
             return Ok(*index);
         }
 
-        if self.files.len() >= std::u16::MAX as usize {
-            return Err(SymCacheErrorKind::TooManyValues(ValueKind::File).into());
+        // Instead of failing hard when exceeding the maximum allowed number of files (u16::MAX),
+        // we rather insert a sentinel "<unknown>" file and use that for all files that exceed the max.
+        if self.files.len() > u16::MAX as usize {
+            return Ok(u16::MAX);
+        }
+        if self.files.len() == u16::MAX as usize {
+            record = format::FileRecord {
```

Comment:
You overwrite the record here, and down below insert it into `file_cache`. This creates an invalid cache entry. You need to cache the original record, and only make the push into `self.files` conditional.